go-controller/pkg/ovn/master.go

```diff
@@ -371,16 +371,6 @@ func (oc *Controller) syncNodeManagementPort(node *kapi.Node, hostSubnets []*net
 		return err
 	}
 
-	if macAddress == nil {
```

Comment:
@danwinship if for some reason the node removes its node management port MAC address annotation, this is supposed to remove the management port to make sure we clean up the database and dont leave stale ports around.

Reply:
But we never do that. There's not even a function that lets you do that. (`SetNodeManagementPortMACAddress` would set the annotation to the string `"<nil>"` if you passed it a `nil` address.) And why would you want to remove the management port but, eg, leave the node's logical switch there?

Reply:
It's clearly a corner case, it was about consistency if for some reason the annotation got removed. We don't have any kind of periodic reconciliation to do this yet.

Reply:
@dcbw i don't think we have a workflow where someone would delete the management port mac address annotation. also, we don't do anything if someone deletes the l3-gateway-config annotation.

Reply:
Ok, as long as it's consistent with our other annotations... we do need some reconciliation loop here to make sure we clean stuff up, even if unexpected things like annotation deletion happen.